src/frontend/components/SearchBar/SearchBar.js

```diff
@@ -1,48 +1,138 @@
+  }
+];
 
-// End of user objects
+export default class SearchBar extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      languageSearchQuery: "",
+      locationSearchQuery: "",
+      filteredUsers: "",
+      filteredLocation: ""
+    };
+  }
 
-export default function SearchBar() {
-  return (
-    <div className="Search">
-      <input className="Search-language-input search-bar" type="search" placeholder="search language"/>
-      <input className="Search-location-input search-bar" type="search" placeholder="search location"/>
-    </div>
-  );
+  searchDatabase() {
+    let filteredUsers = users
+      .filter(user =>
+        user.language
+          .toLowerCase()
+          .includes(this.state.languageSearchQuery.toLowerCase())
+      )
+      .map(profile => (
+        <SearchResults
+          key={profile.id}
+          name={profile.name}
+          language={profile.language}
+          location={profile.location}
+        />
+      ));
+    let filteredLocation = users
+      .filter(user =>
+        user.location
+          .toLowerCase()
+          .includes(this.state.locationSearchQuery.toLowerCase())
+      )
+      .map(profile => (
+        <SearchResults
+          key={profile.id}
+          name={profile.name}
+          language={profile.language}
+          location={profile.location}
+        />
+      ));
+    if (this.state.languageSearchQuery && this.state.locationSearchQuery) {
+      this.setState({
+        filteredUsers,
+        filteredLocation
+      });
+    } else if (this.state.languageSearchQuery) {
+      this.setState({
+        filteredUsers
+      });
+    } else if (this.state.locationSearchQuery) {
+      this.setState({
+        filteredLocation
+      });
+    } else {
+      return "Please try again";
+    }
+  }
+
+  render() {
+    return (
+      <div className="SearchBar">
+        <input
+          className="search-language-input"
+          type="search"
+          placeholder="search language"
+          onChange={e => this.setState({ languageSearchQuery: e.target.value })}
+        />
+        <input
+          className="search-location-input"
+          type="search"
+          placeholder="search location"
+          onChange={e => this.setState({ locationSearchQuery: e.target.value })}
+        />
+        {/* <h1>{this.state.searchValue}</h1> */}
+        <button onClick={() => this.searchDatabase()}>Search</button>
+        {/* <Link to="/results"></Link> */}
+        <div>
+          <table className="resultsTable">
+            <tbody>
+              <tr>
+                <th>User</th>
+                <th>Programming Language</th>
+                <th>Location</th>
+              </tr>
+            </tbody>
+            {this.state.filteredUsers &&
+              this.state.filteredUsers.map(profile => profile)}
+          </table>
+        </div>
+        <div>{this.state.filteredLocation}</div>
+      </div>
+    );
+  }
```

Comment:
it is not ideal to have too many things in a file . Ideally this should be split into 3 files which we can do it later .Also destructuring the props. May be we can refactor in the next commit if you are out of time